ignite/handlers/base_logger.py

```diff
@@ -130,43 +132,72 @@ def _setup_output_metrics_state_attrs(
                             f"in engine's state metrics: {list(engine.state.metrics.keys())}"
                         )
                         continue
-                    metrics_state_attrs[name] = engine.state.metrics[name]
+                    state_metrics[name] = engine.state.metrics[name]
 
         if self.output_transform is not None:
             output_dict = self.output_transform(engine.state.output)
 
             if not isinstance(output_dict, dict):
                 output_dict = {"output": output_dict}
 
-            metrics_state_attrs.update(output_dict)
+            state_metrics.update(output_dict)
 
         if self.state_attributes is not None:
-            metrics_state_attrs.update({name: getattr(engine.state, name, None) for name in self.state_attributes})
+            state_metrics.update({name: getattr(engine.state, name, None) for name in self.state_attributes})
 
-        metrics_state_attrs_dict: Dict[Any, Union[str, float, numbers.Number]] = OrderedDict()
+        metrics: Dict[Any, Union[str, float, numbers.Number]] = OrderedDict()
 
-        def key_tuple_tf(tag: str, name: str, *args: str) -> Tuple[str, ...]:
-            return (tag, name) + args
+        def concat_tuple(parent_tag: Tuple[str], name: str, *args: str) -> Tuple[str, ...]:
+            return parent_tag + (name, *args)
 
-        def key_str_tf(tag: str, name: str, *args: str) -> str:
-            return "/".join((tag, name) + args)
+        def concat_str(parent_tag: str, name: str, *args: str) -> str:
+            return "/".join((parent_tag, name) + args)
 
-        key_tf = key_tuple_tf if key_tuple else key_str_tf
+        concat_tf = concat_tuple if key_tuple else concat_str
 
-        for name, value in metrics_state_attrs.items():
+        self._compute_tags(
+            concat_tf,
+            log_text,
+            node=state_metrics.items(),
+            parent_tag=(self.tag, ) if key_tuple else self.tag,
+            dest_dict=metrics)
+        return metrics
+
+    @classmethod
+    def _compute_tags(
```

Comment:
You can make this just a private helper method in this submodule, not necessarily a classmethod.